Melee hit chance tooltip

Keshash opened the pull request:
## Additions

Describe new functionality added by your code, e.g.
- Adds a tooltip to melee combat. Displays all relevant probablities when hovering over a pawn and wielding a melee weapon

## Reasoning

Why did you choose to implement things this way, e.g.
- Makes it a lot easier to understand whats going on.


## Alternatives

Describe alternative implementations you have considered, e.g.
- Show ideology darklight bonuses - more clutter?
- There is a phrasing issue: what is referred to as "deflection" here refers to parries, which can either be a parry or a block. However, the stat "counter parry bonus" implies its not applied to blocks, but it's actually applied to all deflections. I think renaming the stat to "counter deflection bonus" would make the most sense, but then it's kinda harder to understand from the first glance.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [ ] (For compatibility patches) ...with and without patched mod loaded
- [ ] Playtested a colony (specify how long)

N7Huntsman reviewed (changes requested):
Per discussion on discord, I think the tooltip with all the info is a bit too cluttered and verbose to be the default. I'd recommend we make the 'to hit %' the only thing that displays be default, with a mod option to display all of the other stats.

Keshash commented:
* Made melee tooltip appear for pawns without weapons
* Added mod option to show more detailed information
* Added learning helper entry to explain enabling tooltip and hint at counterparry bonus
* Made some lines not show up if their chances are 0
* Made melee tooltip show only when drafted
* Changed arguments for bodypart targetting slightly to allow inputting different targets, so that the hit penalty is considered in the tooltip calculations
